static/js/components/tiles/bar_tile.tsx

```diff
@@ -83,9 +85,18 @@ export function BarTile(props: BarTilePropType): JSX.Element {
   }, [props, rawData]);
 
   useEffect(() => {
-    if (barChartData) {
-      draw(props, barChartData);
+    if (!barChartData) {
+      return;
     }
+    const debouncedHandler = _.debounce(() => {
+      draw(props, barChartData);
+    }, RESIZE_DEBOUNCE_INTERVAL_MS);
+    const resizeObserver = new ResizeObserver(debouncedHandler);
+    resizeObserver.observe(chartContainerRef.current);
+    return () => {
+      resizeObserver.unobserve(chartContainerRef.current);
+      debouncedHandler.cancel();
+    };
```

Comment:
Wonder if this can be made as a custom React effect as it's used in many places

Reply:
Something like this? https://usehooks.com/useDebounce/

Reply:
done